packages/forms/src/UIForm/fields/MultiSelectTag/MultiSelectTag.component.js

```diff
@@ -190,14 +190,19 @@ export default class MultiSelectTag extends React.Component {
 			const currentProps = props === undefined ? this.props : props;
 			let suggestions = this.getTitleMap(currentProps)
 				.map(item => ({ value: item.value, title: item.name }))
-				.filter(item => currentProps.value.indexOf(item.value) < 0);
+				.filter(item => !currentProps.value.includes(item.value));
 
 			if (currentValue) {
 				const escapedValue = escapeRegexCharacters(currentValue.trim());
-				const regex = new RegExp(escapedValue, 'i');
-				suggestions = suggestions.filter(item => regex.test(item.title));
-
-				if (!suggestions.length && currentProps.schema.restricted === false) {
+				const isExactMatch = suggestions.some(item =>
+					RegExp(`^${escapedValue}$`, 'i').test(item.value),
```

Comment:
please create the regex outside of the some

Reply:
done